python/ambassador/ir/ir.py

```diff
@@ -135,56 +135,54 @@ def check_deltas(
         to_invalidate: List[str] = []
         invalidate_groups_for: List[str] = []
 
-        # OK. If we don't have a cache, just skip all this crap.
-        if cache is not None:
-            # We have a cache. Start by assuming that we'll need to reset it,
+        # OK. If we don't have a cache and there are no deltas, just skip all this crap.
+        if (cache and fetcher.deltas) is not None:
```

Comment:
(Restoring the comment that @the-wondersmith accidentally deleted 😂):

This logic feels a little weird – overall I think you're going for 'if we have a cache and there are deltas", and I think that'd be better expressed as

```
if (cache is not None) and fetcher.deltas:
```

My reasoning is that `fetcher.deltas` actually cannot ever be `None` – it's either an empty list or a list with some things in it.